Sources/Plasma/NucleusLib/pnAsyncCoreExe/pnAceThread.cpp

```diff
@@ -67,6 +84,36 @@ void ThreadDestroy (unsigned exitThreadWaitMs) {
 }
 
 //============================================================================
+AsyncThreadRef AsyncThreadCreate (
+    std::function<void()>    threadProc
+                               ) {
+    AsyncThreadRef ref;
+    ref.impl                  = std::make_shared<AsyncThread>();
+    ref.impl->proc            = threadProc;
+    ref.impl->handle          = nullptr;
+    ref.impl->workTimeMs      = kAsyncTimeInfinite;
+    
+    auto completion = std::make_shared<std::timed_mutex>();
+    completion->lock();
+    ref.impl->completion = completion;
+    
+    auto handle = std::make_shared<std::thread>(&CreateThreadProc, ref);
+    ref.impl->handle = handle;
+    return ref;
+}
+
+void AsyncThreadTimedJoin(AsyncThreadRef& ref, unsigned timeoutMs)
+{
+    bool threadFinished = ref.impl->completion->try_lock_for(std::chrono::milliseconds(timeoutMs));
+    if (threadFinished) {
+        //thread is finished, safe to join with no deadlock risk
+        ref.impl->completion->unlock();
+        ref.impl->handle->join();
+    } else {
+        ref.impl->handle->detach();
+    }
+}
```

Comment:
I've added a test implementation of restoring the original thread wrapper - more or less. Few advantages:
- Real, cross platform async join is working.
- Stuff like PerfAddCounter goes back into the wrapper, each thread body doesn't need to manage that itself.

I've only implemented the thread wrapper so far on the async socket. It seems like it's working fine, although I still need to try Linux/Windows. On macOS async join is now working.

Mostly looking for feedback here - I know this code was removed, but I think there are advantages to wrapping (like async join.) I don't know if stylistically it's all correct.

Looks like there may also be a build issue, I'll look.